lib/vim.c

```diff
@@ -0,0 +1,335 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#include <unistd.h>
+#include <assert.h>
+
+#include "vim.h"
+
+
+#define LOG(x)
+//#define LOG(x) printf(x)
```

Comment:
you may want to use fprintf(stderr, x) instead

Reply:
i removed the logging in 9aee56c3. I just used it for testing in the beginning.